helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/ReportingZooKeeperClient.java

```diff
@@ -51,122 +51,86 @@ public ReportingZooKeeperClient(final ZooKeeperClient client,
 
   @Override
   public void ensurePath(String path) throws KeeperException {
-    try {
+    reporter.time(tag, "ensurePath", (ZooKeeperCallable<Void>) () -> {
       client.ensurePath(path);
-    } catch (KeeperException e) {
-      reporter.checkException(e, tag, "ensurePath");
-      throw e;
-    }
+      return null;
+    });
   }
 
   @Override
   public void ensurePath(String path, boolean excludingLast) throws KeeperException {
-    try {
+    reporter.time(tag, "ensurePath", (ZooKeeperCallable<Void>) () -> {
```

Comment:
I don't think these casts to `(ZooKeeperCallable<Void>)` are necessary; or at least they seem to be in some spots in the diff and not others.